portfolio/src/main/java/com/google/sps/agents/Weather.java

```
@@ -1,61 +1,52 @@
 package com.google.sps.agents;
- 
+
 // Imports the Google Cloud client library
-import com.google.cloud.dialogflow.v2.QueryInput;
-import com.google.cloud.dialogflow.v2.QueryResult;
-import com.google.protobuf.Struct;
 import com.google.protobuf.Value;
-import com.google.sps.agents.Agent;
-import com.google.sps.data.Location;
-import com.google.sps.data.Output;
 import com.google.sps.utils.LocationUtils;
-import java.io.IOException;
-import java.util.ArrayList;
 import java.util.Map;
- 
-/**
- * Weather Agent
- */
+
+/** Weather Agent */
 public class Weather implements Agent {
-    private final String intentName;
-    private String displayAddress;
-    private String searchAddress;
-    private String output = null;
-    
-    public Weather(String intentName, Map<String, Value> parameters) {
-        this.intentName = intentName;
-        try {
-            setParameters(parameters);
-        } catch (Exception e) {
-            return;
-        }
+  private final String intentName;
+  private String displayAddress;
+  private String searchAddress;
+  private String output = null;
+  private String redirect = null;
+
+  public Weather(String intentName, Map<String, Value> parameters) {
+    this.intentName = intentName;
+    try {
+      setParameters(parameters);
+    } catch (Exception e) {
+      return;
     }
+  }
```

sorry, i'm just now noticing that this is the constructor. in general, exceptions in constructors are a bad mix because you get into this weird position of not knowing if construction succeed.

java is probably fighting against us here (only have 1 return value, can't really initialize through an out parameter). it seems that the java-specific way of handling this problem is to let the constructor throw (AFAICT from a quick google); i.e. it's the only way to actually propagate the fact that there was an error.

---

i'm ok with approving this change as is and letting you deal with this in a follow up change